cli/command/image/build/context.go

```diff
@@ -197,17 +209,28 @@ func getDockerfileRelPath(givenContextDir, givenDockerfile string) (absContextDi
 	if !isUNC(absContextDir) {
 		absContextDir, err = filepath.EvalSymlinks(absContextDir)
 		if err != nil {
-			return "", "", errors.Errorf("unable to evaluate symlinks in context path: %v", err)
+			return "", errors.Errorf("unable to evaluate symlinks in context path: %v", err)
 		}
 	}
 
 	stat, err := os.Lstat(absContextDir)
 	if err != nil {
-		return "", "", errors.Errorf("unable to stat context directory %q: %v", absContextDir, err)
+		return "", errors.Errorf("unable to stat context directory %q: %v", absContextDir, err)
 	}
 
 	if !stat.IsDir() {
-		return "", "", errors.Errorf("context must be a directory: %s", absContextDir)
+		return "", errors.Errorf("context must be a directory: %s", absContextDir)
+	}
+	return absContextDir, err
+}
+
+// getDockerfileRelPath returns the dockerfile path relative to the context
+// directory
+func getDockerfileRelPath(absContextDir, givenDockerfile string) (string, error) {
+	var err error
+
+	if givenDockerfile == "-" {
+		return givenDockerfile, nil
```

Comment:
You're correct that is what the old code would return.

If the dockerfile is set to stdin we override it with a random name later on, so returning a fake name here felt confusing. 

I believe the old behaviour was added in an earlier revision of the "dockerfile from stdin" branch where we would use the canonical name instead of a random name.